src/db_impl.cc

```diff
@@ -1132,12 +1182,16 @@ void TitanDBImpl::OnCompactionCompleted(
         if (file->NoLiveData()) {
           edit.DeleteBlobFile(file->file_number(),
                               db_impl_->GetLatestSequenceNumber());
+          continue;
         } else if (static_cast<int>(file->file_level()) >=
                        cf_options.num_levels - 2 &&
                    file->GetDiscardableRatio() >
                        cf_options.blob_file_discardable_ratio) {
           file->FileStateTransit(BlobFileMeta::FileEvent::kNeedMerge);
```

Comment:
It's not related to this PR, but I have a question on how to make sure the file can be deleted by level range.
Cause `TableBuilder::Add` gives no chance for deleted entries, a blob file with a lot of deleted entries may never be checked.

Reply:
@Connor1996 
Sorry for delay.
Live data is not checked by TableBuilder::Add but BlobFileSizeCollector in OnCompactionCompleted(), it compares SSTable referenced value size before/after compaction, so deleted entries will be checked while compacting the deletion record with previous record.